src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -0,0 +1,132 @@
+    {
+        SAFEARRAY* pReturnVal;
+        THROW_IF_FAILED(_uiaProvider.GetSelection(&pReturnVal));
+        return SafeArrayToComArray(pReturnVal);
+    }
+
+    winrt::com_array<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> TermControlAutomationPeer::GetVisibleRanges()
+    {
+        SAFEARRAY* pReturnVal;
+        THROW_IF_FAILED(_uiaProvider.GetVisibleRanges(&pReturnVal));
+        return SafeArrayToComArray(pReturnVal);
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider TermControlAutomationPeer::RangeFromChild(Windows::UI::Xaml::Automation::Provider::IRawElementProviderSimple childElement)
+    {
+        ::ITextRangeProvider* returnVal;
+        // ScreenInfoUiaProvider doesn't actually use parameter, so just pass in nullptr
+        _uiaProvider.RangeFromChild(/* IRawElementProviderSimple */ nullptr,
+                                    &returnVal);
+
+        auto parentProvider = this->ProviderFromPeer(*this);
+        auto xutr = winrt::make_self<XamlUiaTextRange>(returnVal, parentProvider);
+        return xutr.as<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>();
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider TermControlAutomationPeer::RangeFromPoint(Windows::Foundation::Point screenLocation)
+    {
+        ::ITextRangeProvider* returnVal;
+        _uiaProvider.RangeFromPoint({ screenLocation.X, screenLocation.Y }, &returnVal);
+
+        auto parentProvider = this->ProviderFromPeer(*this);
+        auto xutr = winrt::make_self<XamlUiaTextRange>(returnVal, parentProvider);
+        return xutr.as<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>();
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider TermControlAutomationPeer::DocumentRange()
+    {
+        ::ITextRangeProvider* returnVal;
+        _uiaProvider.get_DocumentRange(&returnVal);
+
+        auto parentProvider = this->ProviderFromPeer(*this);
+        auto xutr = winrt::make_self<XamlUiaTextRange>(returnVal, parentProvider);
+        return xutr.as<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>();
+    }
+
+    Windows::UI::Xaml::Automation::SupportedTextSelection TermControlAutomationPeer::SupportedTextSelection()
+    {
+        ::SupportedTextSelection returnVal;
+        _uiaProvider.get_SupportedTextSelection(&returnVal);
+        return static_cast<Windows::UI::Xaml::Automation::SupportedTextSelection>(returnVal);
+    }
+
+#pragma endregion
+
+    RECT TermControlAutomationPeer::GetBoundingRectWrapped()
+    {
+        auto rect = GetBoundingRectangle();
+        return {
+            gsl::narrow<LONG>(rect.X),
+            gsl::narrow<LONG>(rect.Y),
+            gsl::narrow<LONG>(rect.X + rect.Width),
+            gsl::narrow<LONG>(rect.Y + rect.Height)
+        };
+    }
+
+    winrt::com_array<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> TermControlAutomationPeer::SafeArrayToComArray(SAFEARRAY* textRanges)

```

Comment:
Could we get a doc comment explaining why this is necessary? I'm like a yellow belt at COM so I have no idea what I'm looking at.

Reply:
Agreed, given this was a mental conundrum when we discussed it, can you put a lot of comments in here saying why each step is necessary and for what (in addition to perhaps doc-ing the method).